beacon-chain/rpc/beaconv1/blocks.go

```diff
@@ -70,10 +70,14 @@ func (bs *Server) ListBlockHeaders(ctx context.Context, req *ethpb.BlockHeadersR
 			return nil, status.Errorf(codes.Internal, "Could not retrieve blocks: %v", err)
 		}
 	} else {
-		blks, blkRoots, err = bs.BeaconDB.Blocks(ctx, filters.NewFilter().SetStartSlot(req.Slot).SetEndSlot(req.Slot))
+		_, blks, err = bs.BeaconDB.BlocksBySlot(ctx, req.Slot)
```

Comment:
In the event no blocks exist for that slot, we will instead return a zero length slice for blocks rather than a nil object, so it ends up skipping the check below
```go
if blks == nil {
.
.
.
```

Reply:
It seems to be of same behaviour as before, for the `bs.BeaconDB.Blocks` it will also return a zero length slice. Can we consider this check is redundant and is safe to be removed? But it might be a bit out of this PR scope.

Reply:
Actually the `Blocks` and `BlockRoots` methods are of a slightly different behaviour as `BlockRoots` has an extra check:

```
if err != nil {
  return nil, errors.Wrap(err, "could not retrieve block roots")
}
```
though `err != nil` for both cases.

But it means the slice returned can potentially be `nil` for `BlockRoots` but not `Blocks`. I followed the same behaviour in `BlocksBySlot` and `BlockRootsBySlot`. Maybe we should align that (at least for the new getters), e.g. always return empty slice instead of `nil`?

Reply:
Ah k, that is fair. Maybe for completeness sake just add this in 
```go
if blks == nil || len(blks) == 0 {
```
Its a minor change to add in here

Reply:
It will alter the current behaviour of `ListBlockHeaders`, from returning an empty `BlockHeadersResponse` to `err`. Is it fine to make the changes?

Reply:
Given that we explicitly use this grpc error code in the method 
```
status.Error(codes.NotFound, "Could not find requested blocks")
```
in the event of a nil `blocks` slice, I am going to assume it is an oversight that we currently return an empty BlockHeadersResponse when no blocks exist. I will let @terencechain chime in here since he has a better idea on our API response patterns.

Reply:
I think we should not alter the API behaviour in this PR. We can have a separate one if it is confirmed to be any issue. What do you think @terencechain?

Reply:
Separate one is better. Thank you